libraries/Microsoft.Bot.Builder.Dialogs/SkillDialog.cs

```diff
@@ -197,5 +209,70 @@ private async Task<Activity> SendToSkillAsync(ITurnContext context, Activity act
 
             return eocActivity;
         }
+
+        private async Task<bool> InterceptOAuthCardsAsync(ITurnContext turnContext, Activity activity, string connectionName, CancellationToken cancellationToken)
+        {
+            if (activity?.Attachments != null && !string.IsNullOrWhiteSpace(connectionName))
+            {
+                var tokenExchangeProvider = turnContext.Adapter as IExtendedUserTokenProvider;
+                if (tokenExchangeProvider == null)
+                {
+                    // The adapter may choose not to support token exchange, in which case we fallback to showing an oauth card to the user.
+                    return false;
+                }
```

Comment:
The if statement on lines 215 and 218 both abort this method call, otherwise the bot won't attempt to intercept Skill-sent OAuthCards.

Further below, if we don't find an actual OAuthCard, the bot stops attempting to intercept as well.

Combining all of these cases we could refactor it to something like the following:

```suggestion
           var oAuthCard = activity?.Attachments.FirstOrDefault(a => a?.ContentType == OAuthCard.ContentType)
           
           // There are three scenarios where the parent cannot intercept OAuthCards:
           // 1. There are no OAuthCards to intercept.
           // 2. No connection name was provided.
           // 3. The adapter may choose not to support token exchange, in which case we fallback to showing the Skill's OAuthCard to the user.
            if (oAuthCard!= null && !string.IsNullOrWhiteSpace(connectionName) && turnContext.Adapter is IExtendedUserTokenProvider tokenExchangeProvider)
            {
```

Reply:
@gabog and I synced on this and did additional refactoring in an upcoming PR, resolved.